homeassistant/components/zha/core/helpers.py

```diff
@@ -136,3 +139,36 @@ def convert_ieee(ieee_str):
     """Convert given ieee string to EUI64."""
     from zigpy.types import EUI64, uint8_t
     return EUI64([uint8_t(p, base=16) for p in ieee_str.split(':')])
+
+
+def construct_unique_id(cluster):
+    """Construct a unique id from a cluster."""
+    ieee = cluster.endpoint.device.ieee
+    ieeetail = ''.join(['%02x' % (o, ) for o in ieee[-4:]])
+    manufacturer = cluster.endpoint.manufacturer
+    model = cluster.endpoint.model
+    unique_id = None
+    if manufacturer and model is not None:
+        unique_id = "{}_{}_{}_{}_{}".format(
+            slugify(manufacturer),
```

Comment:
why would we include manufacturer and model in unique ID? Could they otherwise overlap?

Reply:
Not really. This keeps them in line with the existing entity Id structure that is already in place.

Reply:
I say we leave the `unique_id` based on IEEE + Endpoint or IEEE + Endpoint + Cluster_ID. My opinion: it is fine for entity_id to include manufacturer/model etc, but unique ID should be left as it was prior. 
And it also going to be  a big breaking change.  I'd rather rename this into `construct_entity_id()`

Reply:
I can do ieee + endpoint + cluster 

To be clear this is currently only used in listeners (later PR). I did it this way to align them with entity ids so the logic could be shared in the future but I can def change to what is suggested above.

Reply:
Nope. Was just making sure the intent was clear. I’ll align on that it will make our lives a lot easier!